Lib/doctest.py

```diff
@@ -1395,11 +1403,12 @@ def __run(self, test, compileflags, out):
                 exec(compile(example.source, filename, "single",
                              compileflags, True), test.globs)
                 self.debugger.set_continue() # ==== Example Finished ====
-                exception = None
+                exc_info = None
             except KeyboardInterrupt:
                 raise
             except:
-                exception = sys.exc_info()
+                exc_info = sys.exc_info()
+                exc_info = (*exc_info[:2], exc_info[2].tb_next)
```

Comment:
We don't need to use sys here:

```
except BaseException as e:
    exc_info = type(e), e, e.__traceback__.tb_next
```

Reply:
Alternatively, save just the exception here and access its parts below where you need them.

Reply:
It does not work, because a traceback here is a not a real traceback. You cannot set it as the part of the exception. Fortunately, all traceback formatting functions and TestCase.addSubTest() work with the exc_info triple.

Reply:
I applied your suggestion there.